### PR TITLE
Segment Replication - Update /_cat/segment_replication API with backpressure metrics.

### LINKED ISSUE #4478: [Segment Replication & Remote Translog] Back-pressure and Recovery for lagging replica copies
**Is your feature request related to a problem? Please describe.**
Once we enable [segment based replication](https://github.com/opensearch-project/OpenSearch/issues/2229) for an index, we wouldn't be indexing any operation on the replica(just writing to translog for durability). Just by virtue of having a successful write to translog we would assume that the replica is caught up. However, since no indexing operation is applied on replicas except the segments on checkpoint refresh, it's possible that the replica may not have successfully processed the checkpoint for a while due to shard overload/slow I/O would still be serving reads. 
Currently there are no additional mechanisms(once translog has been written on the replica) to apply back pressure on primary if the replica is slow in processing checkpoints which would be aggravated with remote translog since there wouldn't be any I/O on replica at all since remote translog writes on primary will handle durability altogether.

**Describe the solution you'd like**
Need to support mechanisms to apply back pressure and as a last resort fail the replica copy if its unable to process any further checkpoint beyond a threshold

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by mch2**
### Description
This change updates the existing /_cat/segment_replication API to include backpressure metrics. It does this by returning stats from primary shards for its tracked replication group and merging it with metrics returned from replicas. Primary captured  metrics will now appear by default, with replica per sync events showing when detailed=true is set.

### Issues Resolved
#4478

### Check List
- [x] New functionality includes testing.
  - [x] All tests pass
- [x] New functionality has been documented.
  - [x] New functionality has javadoc added
- [x] Commits are signed per the DCO using --signoff
- [x] Commit changes are listed out in CHANGELOG.md file (See: [Changelog](../blob/main/CONTRIBUTING.md#changelog))

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by mch2** [comment]
Whitesource scan will be fixed with https://github.com/opensearch-project/OpenSearch/pull/6629

**Review by Rishikesh1159 — APPROVED** [review]

File: server/src/main/java/org/opensearch/index/SegmentReplicationShardStats.java [hunk 1]
```diff
@@ -96,17 +115,18 @@ public void writeTo(StreamOutput out) throws IOException {
     @Override
     public String toString() {
         return "SegmentReplicationShardStats{"
-            + "nodeId='"
-            + nodeId
-            + '\''
+            + "allocationId='"
```
Comment (dreamer-89) [inline server/src/main/java/org/opensearch/index/SegmentReplicationShardStats.java:118]: + "allocationId="

File: server/src/main/java/org/opensearch/rest/action/cat/RestCatSegmentReplicationAction.java [hunk 2]
```diff
@@ -135,58 +140,61 @@ public Table buildSegmentReplicationTable(RestRequest request, SegmentReplicatio
...
-                    } else if (id1 > id2) {
-                        return 1;
-                    } else {
-                        return 0;
-                    }
-                }
+            CollectionUtil.introSort(replicationPerGroupStats, (o1, o2) -> {
+                int id1 = o1.getShardId().id();
+                int id2 = o2.getShardId().id();
+                return Integer.compare(id1, id2);
```
Comment (dreamer-89) [inline server/src/main/java/org/opensearch/rest/action/cat/RestCatSegmentReplicationAction.java:154]: 👍

File: server/src/internalClusterTest/java/org/opensearch/indices/replication/SegmentReplicationStatsIT.java [hunk 3]
```diff
@@ -120,27 +129,228 @@ public void testSegmentReplicationStatsResponseForActiveAndCompletedOnly() throw
...
+            assertEquals(2, replicationStats.size());
+            final List<SegmentReplicationPerGroupStats> replicationPerGroupStats = replicationStats.get(INDEX_NAME);
+            assertEquals(1, replicationPerGroupStats.size());
+            final SegmentReplicationPerGroupStats perGroupStats = replicationPerGroupStats.get(0);
+            assertEquals(perGroupStats.getShardId(), indexShard.shardId());
+            final Set<SegmentReplicationShardStats> replicaStats = perGroupStats.getReplicaStats();
+            assertEquals(1, replicaStats.size());
+            for (SegmentReplicationShardStats replica : replicaStats) {
+                assertNotNull(replica.getCurrentReplicationState());
+            }
```
[the remark this answers is not in the document]
Comment (mch2) [reply server/src/internalClusterTest/java/org/opensearch/indices/replication/SegmentReplicationStatsIT.java]: yes this is missing, added.

File: server/src/internalClusterTest/java/org/opensearch/indices/replication/SegmentReplicationStatsIT.java [hunk 4]
```diff
@@ -120,27 +129,228 @@ public void testSegmentReplicationStatsResponseForActiveAndCompletedOnly() throw
...
+        });
+    }
+
+    public void testGetSpecificShard() throws Exception {
+        internalCluster().startClusterManagerOnlyNode();
+        List<String> nodes = new ArrayList<>();
+        final String primaryNode = internalCluster().startNode();
+        nodes.add(primaryNode);
+        nodes.add(internalCluster().startNode());
```
[the remark this answers is not in the document]
Comment (mch2) [reply server/src/internalClusterTest/java/org/opensearch/indices/replication/SegmentReplicationStatsIT.java]: yep you are right, cleaned this up & the other tests where this is out of order.

**Pull request merged**